x-pack/plugin/kql/src/main/antlr/KqlBase.g4

```diff
@@ -26,9 +26,9 @@ topLevelQuery
     ;
 
 query
-    : query (AND | OR) query          #booleanQuery
-    | NOT subQuery=simpleQuery        #notQuery
-    | simpleQuery                     #defaultQuery
+    : <assoc=right> query (AND | OR) query     #booleanQuery
```

Comment:
ℹ️  Changed the grammar because operators precedence tests surfaced some errors.